cmd/collectors/zapi/plugins/shelf/shelf.go

```diff
@@ -431,20 +439,19 @@ func (my *Shelf) handleCMode(result *node.Node) ([]*matrix.Matrix, error) {
 	return output, nil
 }
 
-func (my *Shelf) handle7Mode(result *node.Node) ([]*matrix.Matrix, error) {
+func (my *Shelf) handle7Mode(result []*node.Node) ([]*matrix.Matrix, error) {
```

Comment:
Yes.
Earlier we got zapi response from `InvokeRequestString` and then we parse in this cmode/7mode function. But now, we parse and get all object records in array form in `InvokeZapiCall`.

Reply:
little correction:
cmode: we parse and get all object records in array form in InvokeZapiCall.
7mode: we return whole zapi response in array with 1 record in InvokeZapiCall. because in 7mode, we need to parse channel first and then shelves.